marlowe-playground-client/src/MarloweEditor/BottomPanel.purs

```diff
@@ -1,66 +1,43 @@
-import Pretty (showPrettyToken)
-import Servant.PureScript.Ajax (AjaxError(..), ErrorDescription(..))
+import Halogen.HTML.Properties (class_, classes)
+import MarloweEditor.Types (Action(..), BottomPanelView(..), State, _editorErrors, _editorWarnings, _showErrorDetail, contractHasErrors)
+import StaticAnalysis.BottomPanel (analysisResultPane, analyzeButton)
+import StaticAnalysis.Types (_analysisState, isCloseAnalysisLoading, isReachabilityLoading, isStaticLoading)
 import Text.Parsing.StringParser.Basic (lines)
 
-isStaticLoading :: AnalysisState -> Boolean
-isStaticLoading (WarningAnalysis remoteData) = isLoading remoteData
-
-isStaticLoading _ = false
-
-isReachabilityLoading :: AnalysisState -> Boolean
-isReachabilityLoading (ReachabilityAnalysis (AnalysisInProgress _)) = true
-
-isReachabilityLoading _ = false
-
-isCloseAnalysisLoading :: AnalysisState -> Boolean
-isCloseAnalysisLoading (CloseAnalysis (AnalysisInProgress _)) = true
-
-isCloseAnalysisLoading _ = false
-
 panelContents :: forall p. State -> BottomPanelView -> HTML p Action
 panelContents state StaticAnalysisView =
   section
     [ classes [ ClassName "panel-sub-header", aHorizontal, Classes.panelContents ]
     ]
     [ analysisResultPane state
-    , button [ onClick $ const $ Just $ AnalyseContract, enabled enabled', classes (if enabled' then [ ClassName "analyse-btn" ] else [ ClassName "analyse-btn", ClassName "disabled" ]) ]
-        [ text (if loading then "Analysing..." else "Analyse for warnings") ]
-    , button [ onClick $ const $ Just $ AnalyseReachabilityContract, enabled enabled', classes (if enabled' then [ ClassName "analyse-btn" ] else [ ClassName "analyse-btn", ClassName "disabled" ]) ]
-        [ text (if loadingReachability then "Analysing..." else "Analyse reachability") ]
-    , button [ onClick $ const $ Just $ AnalyseContractForCloseRefund, enabled enabled', classes (if enabled' then [ ClassName "analyse-btn" ] else [ ClassName "analyse-btn", ClassName "disabled" ]) ]
-        [ text (if loadingCloseAnalysis then "Analysing..." else "Analyse for refunds on Close") ]
+    , analyzeButton loadingAnalyseContract enabled' "Analyse for warnings" AnalyseContract
+    , analyzeButton loadingReachability enabled' "Analyse reachability" AnalyseReachabilityContract
+    , analyzeButton loadingCloseAnalysis enabled' "Analyse for refunds on Close" AnalyseContractForCloseRefund
     ]
   where
-  loading = state ^. _analysisState <<< to isStaticLoading
+  loadingAnalyseContract = state ^. _analysisState <<< to isStaticLoading
 
   loadingReachability = state ^. _analysisState <<< to isReachabilityLoading
 
   loadingCloseAnalysis = state ^. _analysisState <<< to isCloseAnalysisLoading
 
-  enabled' = not loading && not loadingReachability && not loadingCloseAnalysis && not contractHasErrors state
+  enabled' = not loadingAnalyseContract && not loadingReachability && not loadingCloseAnalysis && not contractHasErrors state
```

Comment:
We could call it something like `loadingWarningAnalysis` since we are at it. The original name was worse because when I came up with it it was the only kind of analysis.

Reply:
changed 👍🏻